<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>CONTRIBUTING to variantCell â€¢ variantCell</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css"><script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet"><script src="pkgdown.js"></script><meta property="og:title" content="CONTRIBUTING to variantCell"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-title-body">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">variantCell</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="articles/build_snp_database.html">Building the SNP database</a>
    </li>
    <li>
      <a href="articles/cell_level_DE.html">Cell-level DE SNP Analsyis</a>
    </li>
    <li>
      <a href="articles/donor_recipient.html">Determining Donor and Recipient Cells</a>
    </li>
    <li>
      <a href="articles/group_level_DE.html">Group-level DE SNP Analsyis</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>CONTRIBUTING to variantCell</h1>
    </div>

<div id="contributing-to-variantcell" class="section level1">

<p>Thank you for your interest in contributing to variantCell! This document outlines how you can contribute to this R package for single-cell SNP analysis.</p>
<div class="section level2">
<h2 id="table-of-contents">Table of Contents<a class="anchor" aria-label="anchor" href="#table-of-contents"></a></h2>
<ol style="list-style-type: decimal"><li><a href="#code-of-conduct">Code of Conduct</a></li>
<li><a href="#getting-started">Getting Started</a></li>
<li><a href="#development-workflow">Development Workflow</a></li>
<li><a href="#pull-request-process">Pull Request Process</a></li>
<li><a href="#style-guidelines">Style Guidelines</a></li>
<li><a href="#testing">Testing</a></li>
<li><a href="#issue-reporting">Issue Reporting</a></li>
<li><a href="#feature-requests">Feature Requests</a></li>
<li><a href="#contact">Contact</a></li>
</ol></div>
<div class="section level2">
<h2 id="code-of-conduct">Code of Conduct<a class="anchor" aria-label="anchor" href="#code-of-conduct"></a></h2>
<p>By participating in this project, you agree to uphold our Code of Conduct (we follow the <a href="https://www.contributor-covenant.org/" class="external-link">Contributor Covenant</a>). Please ensure all interactions are respectful, inclusive, and professional.</p>
</div>
<div class="section level2">
<h2 id="getting-started">Getting Started<a class="anchor" aria-label="anchor" href="#getting-started"></a></h2>
<div class="section level3">
<h3 id="prerequisites">Prerequisites<a class="anchor" aria-label="anchor" href="#prerequisites"></a></h3>
<ul><li>R (latest stable version recommended)</li>
<li>Core dependencies: R6, data.table, Matrix, ggplot2, cowplot, GenomicRanges, IRanges, AnnotationHub, matrixStats, ensembldb, circlize, ComplexHeatmap, parallel, doParallel, foreach</li>
<li>Optional: Seurat (for Seurat objects), SingleCellExperiment (for SCE objects)</li>
</ul></div>
<div class="section level3">
<h3 id="fork-and-clone">Fork and Clone<a class="anchor" aria-label="anchor" href="#fork-and-clone"></a></h3>
<ol style="list-style-type: decimal"><li>Fork the repository on GitHub</li>
<li>Clone your fork locally:</li>
</ol><div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">git</span> clone https://github.com/YOUR-USERNAME/variantCell.git</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="bu">cd</span> variantCell</span></code></pre></div>
<ol start="3" style="list-style-type: decimal"><li>Add the original repository as an upstream remote:</li>
</ol><pre><code>bash
git remote add upstream https://github.com/potterae/variantCell.git</code></pre>
<ol start="4" style="list-style-type: decimal"><li>Install the package in development mode:</li>
</ol><pre><code><span><span class="va">bash</span></span>
<span><span class="fu">rdevtools</span><span class="fu">::</span><span class="fu">install_local</span><span class="op">(</span><span class="st">"path/to/variantCell"</span>, force <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="development-workflow">Development Workflow<a class="anchor" aria-label="anchor" href="#development-workflow"></a></h3>
<ol style="list-style-type: decimal"><li>Create a new branch for your feature or bugfix:</li>
</ol><pre><code>bash
git checkout -b feature/your-feature-name</code></pre>
<p>or</p>
<pre><code>bash
git checkout -b fix/issue-you-are-fixing</code></pre>
<ol start="2" style="list-style-type: decimal"><li><p>Make your changes, following the style guidelines below Test your changes with example data (see Testing)</p></li>
<li><p>Test your changes</p></li>
<li><p>Commit your changes with descriptive commit messages</p></li>
</ol><pre><code>bash
git commit -m "Add functionality to analyze donor-specific SNPs"</code></pre>
<ol start="5" style="list-style-type: decimal"><li>Push to your fork:</li>
</ol><pre><code>bash
git push origin feature/your-feature-name</code></pre>
<ol start="6" style="list-style-type: decimal"><li>Submit a pull request (see Pull Request Process)</li>
</ol></div>
<div class="section level3">
<h3 id="pull-request-process">Pull Request Process<a class="anchor" aria-label="anchor" href="#pull-request-process"></a></h3>
<ol style="list-style-type: decimal"><li>Update the README.md and documentation with details of changes if appropriate</li>
<li>Update the version number in relevant files following semantic versioning</li>
<li>Submit your pull request to the main repository</li>
<li>Respond to any feedback or questions during the review process</li>
<li>After approval, a maintainer will merge your PR</li>
</ol></div>
<div class="section level3">
<h3 id="style-guidelines">Style Guidelines<a class="anchor" aria-label="anchor" href="#style-guidelines"></a></h3>
<p>We follow the tidyverse style guide for R code with the following specifics:</p>
<ol style="list-style-type: decimal"><li>Use camelCase for function names within R6 classes, following the existing pattern</li>
<li>Add comprehensive Roxygen documentation for all public functions</li>
<li>Include examples in the documentation</li>
<li>For R6 class methods, follow the documentation format shown in the existing code</li>
<li>Indent with 2 spaces</li>
<li>Maximum line length of 100 characters</li>
<li>Include relevant error handling and input validation</li>
</ol></div>
<div class="section level3">
<h3 id="testing">Testing<a class="anchor" aria-label="anchor" href="#testing"></a></h3>
<ol style="list-style-type: decimal"><li>Create test cases for new functionality</li>
<li>Verify that your code works with example data</li>
<li>If adding new features, provide a small example dataset if possible</li>
<li>Test with both transplant and non-transplant modes if applicable</li>
<li>For performance-critical functions, include benchmarking</li>
</ol><p>Issue Reporting {#issue-reporting}</p>
<p>When reporting issues, please include:</p>
<ol style="list-style-type: decimal"><li>A clear description of the issue</li>
<li>Steps to reproduce the problem</li>
<li>Expected behavior</li>
<li>Actual behavior</li>
<li>R version, package version, and OS information</li>
<li>If possible, a minimal reproducible example</li>
</ol><p>Use the issue templates provided in the repository.</p>
<p>###Feature Requests {#feature-requests}</p>
<p>Feature requests are welcome! Please provide:</p>
<ol style="list-style-type: decimal"><li>A clear description of the proposed feature</li>
<li>The motivation or use case for this feature</li>
<li>If possible, a sketch or example of how it might work</li>
<li>Whether youâ€™re interested in implementing it yourself</li>
</ol><p>Contact {#contact}</p>
<p>For questions about contributing, please:</p>
<p>Open an issue on GitHub Contact the package maintainer: Andrew Potter</p>
<p>Thank you for your interest in improving variantCell!</p>
</div>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>



      <footer><div class="copyright">
  <p></p><p>Developed by Andrew Potter.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

      </footer></div>






  </body></html>

