<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Add a sample with SNP data to the variantCell project — addSampleData • variantCell</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Add a sample with SNP data to the variantCell project — addSampleData"><meta property="og:description" content="Processes and adds a single-cell sample to the variantCell project by integrating metadata,
donor assignments from Vireo, and SNP data from CellSNP. This is the main function
for adding samples to a variantCell project and supports different input data types and
both transplant and non-transplant experimental designs."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">variantCell</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/build_snp_database.html">Building the SNP database</a>
    </li>
    <li>
      <a href="../articles/cell_level_DE.html">Cell-level DE SNP Analsyis</a>
    </li>
    <li>
      <a href="../articles/donor_recipient.html">Determining Donor and Recipient Cells</a>
    </li>
    <li>
      <a href="../articles/group_level_DE.html">Group-level DE SNP Analsyis</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Add a sample with SNP data to the variantCell project</h1>

    <div class="hidden name"><code>addSampleData.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Processes and adds a single-cell sample to the variantCell project by integrating metadata,
donor assignments from Vireo, and SNP data from CellSNP. This is the main function
for adding samples to a variantCell project and supports different input data types and
both transplant and non-transplant experimental designs.</p>
    </div>


    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-sample-id">sample_id<a class="anchor" aria-label="anchor" href="#arg-sample-id"></a></dt>
<dd><p>Character. Unique identifier for the sample.</p></dd>


<dt id="arg-vireo-path">vireo_path<a class="anchor" aria-label="anchor" href="#arg-vireo-path"></a></dt>
<dd><p>Character or NULL. Path to the Vireo donor_ids.tsv file. Required for
transplant mode, can be NULL for non-transplant mode.</p></dd>


<dt id="arg-cellsnp-path">cellsnp_path<a class="anchor" aria-label="anchor" href="#arg-cellsnp-path"></a></dt>
<dd><p>Character. Path to the directory containing CellSNP output files.</p></dd>


<dt id="arg-cell-data">cell_data<a class="anchor" aria-label="anchor" href="#arg-cell-data"></a></dt>
<dd><p>Object. Cell data in one of three forms: Seurat object, SingleCellExperiment
object, or a data frame with cell metadata (with cell identifiers as row names).</p></dd>


<dt id="arg-data-type">data_type<a class="anchor" aria-label="anchor" href="#arg-data-type"></a></dt>
<dd><p>Character. Type of cell_data object: "seurat", "sce", or "dataframe".</p></dd>


<dt id="arg-prefix-text">prefix_text<a class="anchor" aria-label="anchor" href="#arg-prefix-text"></a></dt>
<dd><p>Character. Text to prepend to cell identifiers in the Vireo data to match
the cell barcodes in the input data.</p></dd>


<dt id="arg-donor-type">donor_type<a class="anchor" aria-label="anchor" href="#arg-donor-type"></a></dt>
<dd><p>Named character vector or NULL. Mapping between Vireo donor_id values and
biological roles, e.g. c(donor0 = "Donor", donor1 = "Recipient"). Required for transplant mode.</p></dd>


<dt id="arg-non-transplant-mode">non_transplant_mode<a class="anchor" aria-label="anchor" href="#arg-non-transplant-mode"></a></dt>
<dd><p>Logical. Whether this sample is from a non-transplant experiment
(TRUE) or a transplant experiment (FALSE).</p></dd>


<dt id="arg-min-cells">min_cells<a class="anchor" aria-label="anchor" href="#arg-min-cells"></a></dt>
<dd><p>Integer. Minimum number of cells with alternative allele required for a SNP
to be included in the filtered dataset.</p></dd>


<dt id="arg-min-alt-frac">min_alt_frac<a class="anchor" aria-label="anchor" href="#arg-min-alt-frac"></a></dt>
<dd><p>Numeric. Minimum alternative allele fraction required when counting cells
for the min_cells filter.</p></dd>


<dt id="arg-normalize">normalize<a class="anchor" aria-label="anchor" href="#arg-normalize"></a></dt>
<dd><p>Logical. Whether to calculate normalized SNP counts (TRUE) or not (FALSE).</p></dd>


<dt id="arg-scale-factor">scale.factor<a class="anchor" aria-label="anchor" href="#arg-scale-factor"></a></dt>
<dd><p>Numeric. Scaling factor for normalization. Only used if normalize=TRUE.</p></dd>


<dt id="arg-sample-metadata">sample_metadata<a class="anchor" aria-label="anchor" href="#arg-sample-metadata"></a></dt>
<dd><p>Data frame or NULL. Additional sample-level metadata.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>Invisibly returns self (the variantCell object) with the sample added to the samples list.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>This function performs several steps:</p><ol><li><p>Processes the cell data and donor assignments based on data_type and mode</p></li>
<li><p>Reads and processes the CellSNP data (SNP info and count matrices)</p></li>
<li><p>Matches cell barcodes between the data sources</p></li>
<li><p>Optionally normalizes the count data</p></li>
<li><p>Filters SNPs based on minimum criteria</p></li>
<li><p>Integrates metadata and stores the sample in the project</p></li>
</ol><p>The function supports two modes:</p><ul><li><p>Transplant mode: Uses Vireo to assign cells to donors in transplantation scenarios</p></li>
<li><p>Non-transplant mode: Treats all cells as coming from a single donor</p></li>
</ul></div>
    <div id="note">
    <h2>Note</h2>

<ul><li><p>For transplant mode, vireo_path and donor_type parameters are required</p></li>
<li><p>For non-transplant mode, vireo_path can be NULL and donor_type defaults to c(donor0 = "donor0")</p></li>
<li><p>The <code>normalize</code> parameter controls whether normalized expression values are calculated,
which is useful for expression-based analyses</p></li>
<li><p>The function expects specific file structure for CellSNP output: a base VCF file,
and AD/DP count matrices in Matrix Market format</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Initialize a variantCell project</span></span></span>
<span class="r-in"><span><span class="va">project</span> <span class="op">&lt;-</span> <span class="va"><a href="variantCell-package.html">variantCell</a></span><span class="op">$</span><span class="fu">new</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Example 1: Add a sample in transplant mode using a Seurat object</span></span></span>
<span class="r-in"><span><span class="va">project</span><span class="op">$</span><span class="fu">addSampleData</span><span class="op">(</span></span></span>
<span class="r-in"><span>  sample_id <span class="op">=</span> <span class="st">"transplant_sample1"</span>,</span></span>
<span class="r-in"><span>  vireo_path <span class="op">=</span> <span class="st">"path/to/vireo/donor_ids.tsv"</span>,</span></span>
<span class="r-in"><span>  cellsnp_path <span class="op">=</span> <span class="st">"path/to/cellsnp/output/"</span>,</span></span>
<span class="r-in"><span>  cell_data <span class="op">=</span> <span class="va">seurat_object</span>,</span></span>
<span class="r-in"><span>  data_type <span class="op">=</span> <span class="st">"seurat"</span>,</span></span>
<span class="r-in"><span>  prefix_text <span class="op">=</span> <span class="st">"Patient1_Sample1_"</span>,</span></span>
<span class="r-in"><span>  donor_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>donor0 <span class="op">=</span> <span class="st">"Donor"</span>, donor1 <span class="op">=</span> <span class="st">"Recipient"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  normalize <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Example 2: Add a sample in non-transplant mode using a metadata data frame</span></span></span>
<span class="r-in"><span><span class="va">project</span><span class="op">$</span><span class="fu">addSampleData</span><span class="op">(</span></span></span>
<span class="r-in"><span>  sample_id <span class="op">=</span> <span class="st">"non_transplant_sample1"</span>,</span></span>
<span class="r-in"><span>  vireo_path <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>  cellsnp_path <span class="op">=</span> <span class="st">"path/to/cellsnp/output/"</span>,</span></span>
<span class="r-in"><span>  cell_data <span class="op">=</span> <span class="va">metadata_df</span>,</span></span>
<span class="r-in"><span>  data_type <span class="op">=</span> <span class="st">"dataframe"</span>,</span></span>
<span class="r-in"><span>  prefix_text <span class="op">=</span> <span class="st">"Normal_Sample1_"</span>,</span></span>
<span class="r-in"><span>  non_transplant_mode <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>  normalize <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Andrew Potter.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

      </footer></div>






  </body></html>

