<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>findSNPsByGroup: Group-Level SNP Presence Analysis — findSNPsByGroup • variantCell</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="findSNPsByGroup: Group-Level SNP Presence Analysis — findSNPsByGroup"><meta property="og:description" content="Identifies SNPs that are exclusively or predominantly present in one cell group compared to another.
This function analyzes alternative allele frequencies between groups using aggregated data to detect
group-specific genetic variants."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">variantCell</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/build_snp_database.html">Building the SNP database</a>
    </li>
    <li>
      <a href="../articles/cell_level_DE.html">Cell-level DE SNP Analsyis</a>
    </li>
    <li>
      <a href="../articles/donor_recipient.html">Determining Donor and Recipient Cells</a>
    </li>
    <li>
      <a href="../articles/group_level_DE.html">Group-level DE SNP Analsyis</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>findSNPsByGroup: Group-Level SNP Presence Analysis</h1>

    <div class="hidden name"><code>findSNPsByGroup.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Identifies SNPs that are exclusively or predominantly present in one cell group compared to another.
This function analyzes alternative allele frequencies between groups using aggregated data to detect
group-specific genetic variants.</p>
    </div>


    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-ident-">ident.1<a class="anchor" aria-label="anchor" href="#arg-ident-"></a></dt>
<dd><p>Character. Primary group identity to analyze.</p></dd>


<dt id="arg-ident-">ident.2<a class="anchor" aria-label="anchor" href="#arg-ident-"></a></dt>
<dd><p>Character, optional. Secondary group identity to compare against.
If NULL, compares against all other groups combined.</p></dd>


<dt id="arg-aggregated-data">aggregated_data<a class="anchor" aria-label="anchor" href="#arg-aggregated-data"></a></dt>
<dd><p>List. Output from aggregateByGroup function with required matrices and metadata.</p></dd>


<dt id="arg-min-depth">min_depth<a class="anchor" aria-label="anchor" href="#arg-min-depth"></a></dt>
<dd><p>Integer. Minimum total read depth required for a group to consider a SNP.</p></dd>


<dt id="arg-min-alt-frac">min_alt_frac<a class="anchor" aria-label="anchor" href="#arg-min-alt-frac"></a></dt>
<dd><p>Numeric between 0 and 1. Minimum alternative allele fraction required in a group
for a SNP to be considered present.</p></dd>


<dt id="arg-max-alt-frac-other">max_alt_frac_other<a class="anchor" aria-label="anchor" href="#arg-max-alt-frac-other"></a></dt>
<dd><p>Numeric between 0 and 1. Maximum alternative allele fraction allowed in the
other group for a SNP to be considered absent there.</p></dd>


<dt id="arg-return-all">return_all<a class="anchor" aria-label="anchor" href="#arg-return-all"></a></dt>
<dd><p>Logical. Whether to return all results regardless of significance.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>List containing:</p>
<dl><dt>results</dt>
<dd><p>Data frame of group-specific SNPs with metrics including genomic position,
gene annotation, depth metrics, allele frequencies, and presence classification.</p></dd>

<dt>summary</dt>
<dd><p>List with analysis overview including counts of SNPs present in each group
and parameters used for filtering.</p></dd>

</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>The function identifies SNPs that are present in one group but absent in another by applying
thresholds to alternative allele frequencies. For each SNP, a presence score is calculated
that quantifies the strength of evidence for group-specific presence, considering both the
frequency difference and the read depth.</p>
<p>A SNP is considered "present" in a group when its alternative allele frequency exceeds
<code>min_alt_frac</code> and the read depth exceeds <code>min_depth</code>. It is considered "absent" in the
other group when its alternative allele frequency is below <code>max_alt_frac_other</code> and the
read depth exceeds <code>min_depth</code>.</p>
<p>The presence score formula is:
score = (alt_frac_present - alt_frac_absent) * (depth/min_depth) * (1 - alt_frac_absent/min_alt_frac)</p>
    </div>
    <div id="note">
    <h2>Note</h2>

<ul><li><p>This function operates on pre-aggregated data from <code><a href="aggregateByGroup.html">aggregateByGroup()</a></code> rather than raw SNP data</p></li>
<li><p>Non-transplant mode is automatically detected from the aggregated data parameters</p></li>
<li><p>Results are sorted by presence score, with highest-scoring SNPs listed first</p></li>
</ul></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="aggregateByGroup.html">aggregateByGroup</a></code> for preparing input data
<code><a href="findDESNPs.html">findDESNPs</a></code> for cell-level differential analysis
<code><a href="plotSNPs.html">plotSNPs</a></code> for visualizing the identified SNPs</p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Aggregate SNP data by cell type</span></span></span>
<span class="r-in"><span><span class="va">agg_data</span> <span class="op">&lt;-</span> <span class="va">proj</span><span class="op">$</span><span class="fu">aggregateByGroup</span><span class="op">(</span></span></span>
<span class="r-in"><span>  group_by <span class="op">=</span> <span class="st">"cell_type"</span>,</span></span>
<span class="r-in"><span>  donor_type <span class="op">=</span> <span class="st">"Donor"</span>,</span></span>
<span class="r-in"><span>  use_normalized <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Find T cell-specific SNPs</span></span></span>
<span class="r-in"><span><span class="va">tc_snps</span> <span class="op">&lt;-</span> <span class="va">proj</span><span class="op">$</span><span class="fu">findSNPsByGroup</span><span class="op">(</span></span></span>
<span class="r-in"><span>  ident.1 <span class="op">=</span> <span class="st">"T_cells"</span>,</span></span>
<span class="r-in"><span>  ident.2 <span class="op">=</span> <span class="st">"B_cells"</span>,</span></span>
<span class="r-in"><span>  aggregated_data <span class="op">=</span> <span class="va">agg_data</span>,</span></span>
<span class="r-in"><span>  min_depth <span class="op">=</span> <span class="fl">20</span>,</span></span>
<span class="r-in"><span>  min_alt_frac <span class="op">=</span> <span class="fl">0.25</span>,</span></span>
<span class="r-in"><span>  max_alt_frac_other <span class="op">=</span> <span class="fl">0.05</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Comparing patient groups</span></span></span>
<span class="r-in"><span><span class="va">patient_snps</span> <span class="op">&lt;-</span> <span class="va">proj</span><span class="op">$</span><span class="fu">findSNPsByGroup</span><span class="op">(</span></span></span>
<span class="r-in"><span>  ident.1 <span class="op">=</span> <span class="st">"ACR"</span>,</span></span>
<span class="r-in"><span>  ident.2 <span class="op">=</span> <span class="st">"No_ACR"</span>,</span></span>
<span class="r-in"><span>  aggregated_data <span class="op">=</span> <span class="va">patient_data</span>,</span></span>
<span class="r-in"><span>  min_alt_frac <span class="op">=</span> <span class="fl">0.1</span>,</span></span>
<span class="r-in"><span>  max_alt_frac_other <span class="op">=</span> <span class="fl">0.02</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Andrew Potter.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

      </footer></div>






  </body></html>

