<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>plotSNPs "Visualize SNP Distribution for a Gene" — plotSNPs • variantCell</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="plotSNPs " visualize snp distribution for a gene plotsnps><meta property="og:description" content="Creates genomic visualizations of SNP distribution for a specified gene, showing alternative allele
frequencies across different cell populations. This function generates interactive plots
displaying SNPs along the gene's coordinates with various grouping options."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">variantCell</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/build_snp_database.html">Building the SNP database</a>
    </li>
    <li>
      <a href="../articles/cell_level_DE.html">Cell-level DE SNP Analsyis</a>
    </li>
    <li>
      <a href="../articles/donor_recipient.html">Determining Donor and Recipient Cells</a>
    </li>
    <li>
      <a href="../articles/group_level_DE.html">Group-level DE SNP Analsyis</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>plotSNPs "Visualize SNP Distribution for a Gene"</h1>

    <div class="hidden name"><code>plotSNPs.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Creates genomic visualizations of SNP distribution for a specified gene, showing alternative allele
frequencies across different cell populations. This function generates interactive plots
displaying SNPs along the gene's coordinates with various grouping options.</p>
    </div>


    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-gene">gene<a class="anchor" aria-label="anchor" href="#arg-gene"></a></dt>
<dd><p>Character. Name of the gene to visualize.</p></dd>


<dt id="arg-group-by">group.by<a class="anchor" aria-label="anchor" href="#arg-group-by"></a></dt>
<dd><p>Character, optional. Primary grouping variable from metadata.
If NULL, all cells are treated as one group.</p></dd>


<dt id="arg-split-by">split.by<a class="anchor" aria-label="anchor" href="#arg-split-by"></a></dt>
<dd><p>Character, optional. Secondary grouping variable for within-group comparisons.
Useful for donor/recipient or condition comparisons within cell types.</p></dd>


<dt id="arg-idents">idents<a class="anchor" aria-label="anchor" href="#arg-idents"></a></dt>
<dd><p>Vector, optional. Specific identity values to include. Must be values from group.by column.</p></dd>


<dt id="arg-min-depth">min_depth<a class="anchor" aria-label="anchor" href="#arg-min-depth"></a></dt>
<dd><p>Integer. Minimum read depth required to include a SNP in visualization.</p></dd>


<dt id="arg-min-cells">min_cells<a class="anchor" aria-label="anchor" href="#arg-min-cells"></a></dt>
<dd><p>Integer. Minimum cells per group required to include a group in visualization.</p></dd>


<dt id="arg-min-alt-frac">min_alt_frac<a class="anchor" aria-label="anchor" href="#arg-min-alt-frac"></a></dt>
<dd><p>Numeric between 0 and 1. Minimum alternative allele fraction required to include a SNP.
Set to 0 to show all SNPs regardless of alt fraction.</p></dd>


<dt id="arg-flank-size">flank_size<a class="anchor" aria-label="anchor" href="#arg-flank-size"></a></dt>
<dd><p>Integer. Size of flanking regions (in bp) to include around the gene.</p></dd>


<dt id="arg-plot-density">plot_density<a class="anchor" aria-label="anchor" href="#arg-plot-density"></a></dt>
<dd><p>Logical. Whether to include density distribution plots below the main visualization.</p></dd>


<dt id="arg-data-out">data_out<a class="anchor" aria-label="anchor" href="#arg-data-out"></a></dt>
<dd><p>Logical. If TRUE, returns a data frame with SNP data instead of plots.</p></dd>


<dt id="arg-use-normalized">use_normalized<a class="anchor" aria-label="anchor" href="#arg-use-normalized"></a></dt>
<dd><p>Logical. Whether to use normalized read depth values.</p></dd>


<dt id="arg-color-scheme">color_scheme<a class="anchor" aria-label="anchor" href="#arg-color-scheme"></a></dt>
<dd><p>Vector. Named vector with "low" and "high" colors for alt fraction gradient.</p></dd>


<dt id="arg-point-size-range">point_size_range<a class="anchor" aria-label="anchor" href="#arg-point-size-range"></a></dt>
<dd><p>Numeric vector of length 2. Range of point sizes (minimum and maximum) for depth representation.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>If data_out = FALSE (default): Plot grid with main SNP visualization and optional density plots.
If data_out = TRUE: Data frame containing SNP information for the gene including positions,
alternative allele frequencies, depths, and other metrics for each group.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>This function creates a comprehensive visualization of SNPs within a gene region, showing:</p><ol><li><p>A main plot with:</p><ul><li><p>Gene structure with exons displayed as black rectangles</p></li>
<li><p>SNP positions represented as points along genomic coordinates</p></li>
<li><p>Alternative allele frequencies encoded by color</p></li>
<li><p>Read depth encoded by point size</p></li>
<li><p>Groups and splits organized vertically</p></li>
</ul></li>
<li><p>Optional density plots showing:</p><ul><li><p>Alternative allele fraction distribution</p></li>
<li><p>Read depth distribution (log10 scale)</p></li>
</ul></li>
</ol><p>The function applies filters to ensure only high-quality data is displayed. SNPs with low
depth or low alternative allele frequency can be excluded. Groups with insufficient cells
are also not displayed.</p>
    </div>
    <div id="note">
    <h2>Note</h2>

<ul><li><p>Requires gene annotation data from AnnotationHub (automatically accessed)</p></li>
<li><p>May take longer to generate for genes with many SNPs or when using many groups</p></li>
<li><p>Setting min_alt_frac = 0 shows all SNPs but may make plots very busy</p></li>
<li><p>The data_out parameter is useful for further custom analysis or visualization</p></li>
</ul></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="findDESNPs.html">findDESNPs</a></code> for identifying differentially expressed SNPs
<code><a href="plotSNPHeatmap.html">plotSNPHeatmap</a></code> for heatmap visualization of multiple genes</p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Basic visualization of a gene</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Split by cell type and donor type</span></span></span>
<span class="r-in"><span><span class="va">project</span><span class="op">$</span><span class="fu">plotSNPs</span><span class="op">(</span></span></span>
<span class="r-in"><span>  gene <span class="op">=</span> <span class="st">"IL7R"</span>,</span></span>
<span class="r-in"><span>  group.by <span class="op">=</span> <span class="st">"cell_type"</span>,</span></span>
<span class="r-in"><span>  split.by <span class="op">=</span> <span class="st">"donor_type"</span>,</span></span>
<span class="r-in"><span>  min_alt_frac <span class="op">=</span> <span class="fl">0.1</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Focus on specific cell types</span></span></span>
<span class="r-in"><span><span class="va">project</span><span class="op">$</span><span class="fu">plotSNPs</span><span class="op">(</span></span></span>
<span class="r-in"><span>  gene <span class="op">=</span> <span class="st">"PDCD1"</span>,</span></span>
<span class="r-in"><span>  group.by <span class="op">=</span> <span class="st">"cell_type"</span>,</span></span>
<span class="r-in"><span>  idents <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD4_T"</span>, <span class="st">"CD8_T"</span>, <span class="st">"Treg"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  flank_size <span class="op">=</span> <span class="fl">10000</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Return data frame instead of plot for custom visualization</span></span></span>
<span class="r-in"><span><span class="va">snp_data</span> <span class="op">&lt;-</span> <span class="va">project</span><span class="op">$</span><span class="fu">plotSNPs</span><span class="op">(</span></span></span>
<span class="r-in"><span>  gene <span class="op">=</span> <span class="st">"HLA-DRB1"</span>,</span></span>
<span class="r-in"><span>  group.by <span class="op">=</span> <span class="st">"condition"</span>,</span></span>
<span class="r-in"><span>  data_out <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Andrew Potter.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

      </footer></div>






  </body></html>

