<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Single-Cell SNP Analysis with Focus on Organ Transplant • variantCell</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Single-Cell SNP Analysis with Focus on Organ Transplant">
<meta name="description" content="Analyzes single-cell SNP data with a focus on transplant settings. Works with single-cell genomics workflows to identify, characterize, and visualize SNPs at cellular resolution, enabling researchers to distinguish donor from recipient cells and quantify SNPs that may influence transplant outcomes.">
<meta property="og:description" content="Analyzes single-cell SNP data with a focus on transplant settings. Works with single-cell genomics workflows to identify, characterize, and visualize SNPs at cellular resolution, enabling researchers to distinguish donor from recipient cells and quantify SNPs that may influence transplant outcomes.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">variantCell</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/build_snp_database.html">Building the SNP database</a></li>
    <li><a class="dropdown-item" href="articles/cell_level_DE.html">Cell-level DE SNP Analsyis</a></li>
    <li><a class="dropdown-item" href="articles/donor_recipient.html">Determining Donor and Recipient Cells</a></li>
    <li><a class="dropdown-item" href="articles/group_level_DE.html">Group-level DE SNP Analsyis</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="variantcell-010---alpha-in-development">variantCell 0.1.0 - alpha (in development)<a class="anchor" aria-label="anchor" href="#variantcell-010---alpha-in-development"></a>
</h1></div>
<p><a href="https://opensource.org/licenses/MIT" class="external-link"><img src="https://img.shields.io/badge/License-MIT-yellow.svg" alt="License: MIT"></a> <a href=""><img src="https://img.shields.io/badge/Status-In%20Development-blue" alt="Status: In Development"></a></p>
<p>A tool for analyzing single-cell SNP data with focus on organ transplant.</p>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>variantCell is an R-based computational tool designed to analyze single-cell SNP data with a particular focus on transplant settings. It works with widely-used single-cell genomics workflows to identify, characterize, and visualize SNPs at cellular resolution, enabling researchers to distinguish donor from recipient cells and quantify SNPs that may influence transplant outcomes.</p>
<p>The package provides a comprehensive framework for processing CellSNP and Vireo output, normalizing SNP counts, and performing differential SNP analysis across cell types or clinical conditions. With built-in visualization capabilities, users can explore variant distributions within genes of interest and create SNP heatmaps across cell clusters.</p>
<p>Unlike existing approaches that analyze variants at the bulk tissue level, variantCell leverages single-cell resolution to reveal cell type-specific genetic signals and heterogeneity within transplanted tissues. This approach offers unique insights into the genetic factors driving transplant rejection, immune response variation, and cellular chimerism in complex tissues, with applications extending to cancer, development, and other areas where genetic mosaicism is biologically significant.</p>
</div>
<div class="section level2">
<h2 id="core-dependencies">Core Dependencies<a class="anchor" aria-label="anchor" href="#core-dependencies"></a>
</h2>
<p>R6, data.table, Matrix, ggplot2, cowplot, GenomicRanges, IRanges, AnnotationHub, matrixStats, ensembldb, circlize, ComplexHeatmap</p>
</div>
<div class="section level2">
<h2 id="optional-dependencies">Optional Dependencies<a class="anchor" aria-label="anchor" href="#optional-dependencies"></a>
</h2>
<p>Seurat: If using Seurat objects as input (not required if using data frames), SingleCellExperiment: If using SingleCellExperiment objects as input Parallel processing findDESNPs function: parallel, doParallel, foreach</p>
</div>
<div class="section level2">
<h2 id="getting-started">Getting Started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h2>
<p>To add samples to the SNP database, the package requires: - An output directory from CellSNP along with cell metadata (as a Seurat object, dataframe, or SingleCellExperiment) - Cell prefixes can be optionally specified when adding sample data to match cell IDs to integrated data</p>
<p>For transplant mode, you’ll also need: - A Vireo output directory - Specification of which donors are ‘Recipient’ and ‘Donor’ (can be determined using the <code>process_vireo</code> helper function)</p>
<p>After adding samples, a SNP database can be built using the <code>buildSNPDatabase</code> function, enabling:</p>
<ul>
<li><p>Finding SNPs present or absent in specific groups</p></li>
<li><p>Performing cell (or group)-level differential SNP expression analysis (uses parallel processing)</p></li>
</ul>
<p>SNPs can be visualized:</p>
<ul>
<li><p>Plotting SNPs for specific genes</p></li>
<li><p>Creating heatmaps of SNPs within cell clusters / groups</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Install from GitHub</p>
<p>Check if devtools is installed, install if needed</p>
<p><code>if (!requireNamespace("devtools", quietly = TRUE)) { install.packages("devtools") }</code></p>
<p><code>devtools::install_github("potterae/variantCell")</code></p>
</div>
<div class="section level2">
<h2 id="license">License<a class="anchor" aria-label="anchor" href="#license"></a>
</h2>
<p>This project is licensed under the MIT License.</p>
</div>
<div class="section level2">
<h2 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h2>
<p>Funding: Heart Institute, Cincinnati Children’s Hospital Medical Center</p>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>If you use this software, please cite:</p>
<p>Andrew Potter, Don Hayes (in preparation). variantCell: A tool for single-cell SNP analysis. GitHub: <a href="https://github.com/potterae/variantCell" class="external-link uri">https://github.com/potterae/variantCell</a></p>
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CONTRIBUTING.html">Contributing guide</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing variantCell</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Andrew Potter <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-8218-3813" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Andrew Potter.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
