<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Group-level DE SNP Analsyis • variantCell</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Group-level DE SNP Analsyis">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">variantCell</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/build_snp_database.html">Building the SNP database</a></li>
    <li><a class="dropdown-item" href="../articles/cell_level_DE.html">Cell-level DE SNP Analsyis</a></li>
    <li><a class="dropdown-item" href="../articles/donor_recipient.html">Determining Donor and Recipient Cells</a></li>
    <li><a class="dropdown-item" href="../articles/group_level_DE.html">Group-level DE SNP Analsyis</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Group-level DE SNP Analsyis</h1>
                        <h4 data-toc-skip class="author">Andrew
Potter</h4>
            
            <h4 data-toc-skip class="date">2025-05-12</h4>
      

      <div class="d-none name"><code>group_level_DE.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="group-level-snp-analysis">Group-level SNP Analysis<a class="anchor" aria-label="anchor" href="#group-level-snp-analysis"></a>
</h2>
<p>Group level differential expression analysis requires a SNP database
to be built (please see previous tutorials). This guide will walk you
through the process of analyzing SNPs at the group level using the
variantCell package.</p>
<div class="section level3">
<h3 id="exploring-snp-metadata">Exploring SNP Metadata<a class="anchor" aria-label="anchor" href="#exploring-snp-metadata"></a>
</h3>
<p>To view the SNP metadata within your project, you can use the
following command:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>\dontrun{</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>  <span class="co"># Initialize a variantCell project</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>  project <span class="ot">&lt;-</span> variantCell<span class="sc">$</span><span class="fu">new</span>()</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>  </span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>  <span class="co"># In practice, you would add samples and build the database first</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>  <span class="co"># project$addSampleData(...)</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>  <span class="co"># project$buildSNPDatabase()</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>  <span class="co"># project$setProjectIdentity("cell_type")</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># View SNP metadata</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/View.html" class="external-link">View</a></span><span class="op">(</span><span class="va">project</span><span class="op">$</span><span class="va">snp_database</span><span class="op">$</span><span class="va">cell_metadata</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="creating-custom-metadata-variables">Creating Custom Metadata Variables<a class="anchor" aria-label="anchor" href="#creating-custom-metadata-variables"></a>
</h3>
<p>You can create new metadata variables to help with grouping or
visualization:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create a combined patient and donor type variable</span></span>
<span><span class="va">project</span><span class="op">$</span><span class="va">snp_database</span><span class="op">$</span><span class="va">cell_metadata</span><span class="op">$</span><span class="va">patient_donor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span></span>
<span>  <span class="va">project</span><span class="op">$</span><span class="va">snp_database</span><span class="op">$</span><span class="va">cell_metadata</span><span class="op">$</span><span class="va">patient</span>, </span>
<span>  <span class="st">'_'</span>, </span>
<span>  <span class="va">project</span><span class="op">$</span><span class="va">snp_database</span><span class="op">$</span><span class="va">cell_metadata</span><span class="op">$</span><span class="va">donor_type</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="aggregating-snp-counts-by-group">Aggregating SNP Counts by Group<a class="anchor" aria-label="anchor" href="#aggregating-snp-counts-by-group"></a>
</h3>
<p>Prior to analyzing SNP expression differences between groups, we need
to aggregate the individual cell SNP counts into group-level
summaries.</p>
<div class="section level4">
<h4 id="aggregatebygroup-function">aggregateByGroup Function<a class="anchor" aria-label="anchor" href="#aggregatebygroup-function"></a>
</h4>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Function signature and parameters</span></span>
<span><span class="va">project</span><span class="op">$</span><span class="fu">aggregateByGroup</span><span class="op">(</span></span>
<span>  <span class="va">group_by</span>,                <span class="co"># Column name in metadata to use for grouping cells</span></span>
<span>  donor_type <span class="op">=</span> <span class="cn">NULL</span>,       <span class="co"># Specific donor type to analyze (e.g., "Donor" or "Recipient")</span></span>
<span>  min_cells_per_group <span class="op">=</span> <span class="fl">3</span>, <span class="co"># Minimum cells required per group</span></span>
<span>  use_normalized <span class="op">=</span> <span class="cn">TRUE</span>    <span class="co"># Whether to use normalized counts</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="example-aggregating-by-disease-status-for-donor-cells">Example: Aggregating by Disease Status for Donor Cells<a class="anchor" aria-label="anchor" href="#example-aggregating-by-disease-status-for-donor-cells"></a>
</h4>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Aggregate SNP counts by ACR status, using only donor cells</span></span>
<span><span class="va">collapsed</span> <span class="op">&lt;-</span> <span class="va">project</span><span class="op">$</span><span class="fu">aggregateByGroup</span><span class="op">(</span></span>
<span>  group_by <span class="op">=</span> <span class="st">"snpGrade"</span>,     <span class="co"># Group by ACR status</span></span>
<span>  use_normalized <span class="op">=</span> <span class="cn">TRUE</span>,     <span class="co"># Use normalized counts</span></span>
<span>  donor_type <span class="op">=</span> <span class="st">'Donor'</span>       <span class="co"># Restrict to donor cells only</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>When executed, you should see output similar to:</p>
<pre><code>Filtering for donor type: Donor
Total cells before filter: 56676
Cells passing filter: 22096
NA values in donor_type: 1003

Collapsed data by snpGrade:
 - Total cells: 22096
 - Total groups: 2
 - Included groups: 2
 - Filtered groups: 0
 - Using normalized counts: TRUE

Group details:
   group n_cells mean_depth filter_status
1    ACR   12598   3981.279      included
2 No_ACR    9498   7392.881      included</code></pre>
</div>
</div>
<div class="section level3">
<h3 id="finding-differentially-expressed-snps-between-groups">Finding Differentially Expressed SNPs Between Groups<a class="anchor" aria-label="anchor" href="#finding-differentially-expressed-snps-between-groups"></a>
</h3>
<p>After aggregating our data, we can identify SNPs that are
differentially present between groups. The <code>findSNPsByGroup</code>
function compares alternative allele frequencies between groups to find
variants that are specifically present in one group but not the
other.</p>
<div class="section level4">
<h4 id="findsnpsbygroup-function">findSNPsByGroup Function<a class="anchor" aria-label="anchor" href="#findsnpsbygroup-function"></a>
</h4>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Function signature and parameters</span></span>
<span><span class="va">project</span><span class="op">$</span><span class="fu">findSNPsByGroup</span><span class="op">(</span></span>
<span>  <span class="va">ident.1</span>,                 <span class="co"># Primary group identity to analyze</span></span>
<span>  ident.2 <span class="op">=</span> <span class="cn">NULL</span>,          <span class="co"># Secondary group identity to compare against (NULL = all others)</span></span>
<span>  <span class="va">aggregated_data</span>,         <span class="co"># Output from aggregateByGroup function</span></span>
<span>  min_depth <span class="op">=</span> <span class="fl">10</span>,          <span class="co"># Minimum read depth required for a group</span></span>
<span>  min_alt_frac <span class="op">=</span> <span class="fl">0.2</span>,      <span class="co"># Minimum alt allele fraction in primary group</span></span>
<span>  max_alt_frac_other <span class="op">=</span> <span class="fl">0.1</span>, <span class="co"># Maximum alt fraction allowed in other group</span></span>
<span>  return_all <span class="op">=</span> <span class="cn">TRUE</span>        <span class="co"># Whether to return all results or only significant ones</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="example-finding-snps-between-acr-and-no_acr-groups">Example: Finding SNPs Between ACR and No_ACR Groups<a class="anchor" aria-label="anchor" href="#example-finding-snps-between-acr-and-no_acr-groups"></a>
</h4>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Find SNPs present in ACR patients but absent in No_ACR patients</span></span>
<span><span class="va">de_results</span> <span class="op">&lt;-</span> <span class="va">project</span><span class="op">$</span><span class="fu">findSNPsByGroup</span><span class="op">(</span></span>
<span>  ident.1 <span class="op">=</span> <span class="st">"ACR"</span>,             <span class="co"># Primary group to analyze</span></span>
<span>  ident.2 <span class="op">=</span> <span class="st">'No_ACR'</span>,          <span class="co"># Comparison group</span></span>
<span>  aggregated_data <span class="op">=</span> <span class="va">collapsed</span>, <span class="co"># Aggregated data from previous step</span></span>
<span>  min_alt_frac <span class="op">=</span> <span class="fl">0.05</span>,         <span class="co"># SNP must have at least 5% alt frequency in ACR</span></span>
<span>  max_alt_frac_other <span class="op">=</span> <span class="fl">0.01</span>    <span class="co"># SNP must have no more than 1% alt frequency in No_ACR</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="examining-the-results">Examining the Results<a class="anchor" aria-label="anchor" href="#examining-the-results"></a>
</h3>
<p>After running the analysis, you can examine the results:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># View summary of the analysis</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">de_results</span><span class="op">$</span><span class="va">summary</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Look at the top SNPs by presence score</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">de_results</span><span class="op">$</span><span class="va">results</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="op">-</span><span class="va">de_results</span><span class="op">$</span><span class="va">results</span><span class="op">$</span><span class="va">presence_score</span><span class="op">)</span>, <span class="op">]</span>, <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># How many SNPs are present in each group?</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">de_results</span><span class="op">$</span><span class="va">results</span><span class="op">$</span><span class="va">presence</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Save results to a CSV file</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">de_results</span><span class="op">$</span><span class="va">results</span>, <span class="st">'SNP_ACR_NoACR_Donor.csv'</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visualizing-snps-in-genes-of-interest">Visualizing SNPs in Genes of Interest<a class="anchor" aria-label="anchor" href="#visualizing-snps-in-genes-of-interest"></a>
</h3>
<p>You can visualize SNPs within specific genes using the
<code>plotSNPs</code> function.</p>
<div class="section level4">
<h4 id="plotsnps-function">plotSNPs Function<a class="anchor" aria-label="anchor" href="#plotsnps-function"></a>
</h4>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Function signature and parameters</span></span>
<span><span class="va">project</span><span class="op">$</span><span class="fu">plotSNPs</span><span class="op">(</span></span>
<span>  <span class="va">gene</span>,                      <span class="co"># Gene symbol to visualize</span></span>
<span>  group.by <span class="op">=</span> <span class="cn">NULL</span>,           <span class="co"># Primary grouping variable from metadata</span></span>
<span>  split.by <span class="op">=</span> <span class="cn">NULL</span>,           <span class="co"># Secondary grouping variable for within-group comparisons</span></span>
<span>  idents <span class="op">=</span> <span class="cn">NULL</span>,             <span class="co"># Specific identity values to include</span></span>
<span>  min_depth <span class="op">=</span> <span class="fl">10</span>,            <span class="co"># Minimum read depth required for a SNP</span></span>
<span>  min_cells <span class="op">=</span> <span class="fl">3</span>,             <span class="co"># Minimum cells per group required</span></span>
<span>  min_alt_frac <span class="op">=</span> <span class="fl">0.2</span>,        <span class="co"># Minimum alternative allele fraction</span></span>
<span>  flank_size <span class="op">=</span> <span class="fl">5000</span>,         <span class="co"># Size of flanking regions (bp) around gene</span></span>
<span>  plot_density <span class="op">=</span> <span class="cn">TRUE</span>,       <span class="co"># Whether to include density plots</span></span>
<span>  data_out <span class="op">=</span> <span class="cn">FALSE</span>,          <span class="co"># Return data frame instead of plots</span></span>
<span>  use_normalized <span class="op">=</span> <span class="cn">FALSE</span>,    <span class="co"># Use normalized read depth values</span></span>
<span>  color_scheme <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"low"</span> <span class="op">=</span> <span class="st">"blue"</span>, <span class="st">"high"</span> <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span>, <span class="co"># Colors for gradient</span></span>
<span>  point_size_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">8</span><span class="op">)</span> <span class="co"># Range of point sizes for depth</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="example-visualizing-jak1-snps-by-acr-status-and-patient-donor">Example: Visualizing JAK1 SNPs by ACR Status and Patient-Donor<a class="anchor" aria-label="anchor" href="#example-visualizing-jak1-snps-by-acr-status-and-patient-donor"></a>
</h4>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot SNPs in JAK1 gene without minimum alt fraction threshold</span></span>
<span><span class="va">project</span><span class="op">$</span><span class="fu">plotSNPs</span><span class="op">(</span></span>
<span>  <span class="st">'JAK1'</span>,</span>
<span>  group.by <span class="op">=</span> <span class="st">'snpGrade'</span>,       <span class="co"># Group by ACR status</span></span>
<span>  split.by <span class="op">=</span> <span class="st">'patient_donor'</span>,  <span class="co"># Split by patient and donor type</span></span>
<span>  min_alt_frac <span class="op">=</span> <span class="fl">0</span>,            <span class="co"># Show all SNPs regardless of alt fraction</span></span>
<span>  plot_density <span class="op">=</span> <span class="cn">FALSE</span>         <span class="co"># Don't show density plots</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="SNP_plot_JAK1.png"></p>
</div>
<div class="section level4">
<h4 id="example-visualizing-with-filtered-snps-and-density-plots">Example: Visualizing With Filtered SNPs and Density Plots<a class="anchor" aria-label="anchor" href="#example-visualizing-with-filtered-snps-and-density-plots"></a>
</h4>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot JAK1 SNPs with minimum alt fraction and density plots</span></span>
<span><span class="va">project</span><span class="op">$</span><span class="fu">plotSNPs</span><span class="op">(</span></span>
<span>  <span class="st">'JAK1'</span>,</span>
<span>  group.by <span class="op">=</span> <span class="st">'snpGrade'</span>,</span>
<span>  split.by <span class="op">=</span> <span class="st">'patient_donor'</span>,</span>
<span>  min_alt_frac <span class="op">=</span> <span class="fl">0.2</span>,         <span class="co"># Only show SNPs with alt fraction ≥ 0.2</span></span>
<span>  plot_density <span class="op">=</span> <span class="cn">TRUE</span>         <span class="co"># Include density plots</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="SNP_plot_JAK1_density.png"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get SNP data for JAK1 as a data frame</span></span>
<span><span class="va">jak1_data</span> <span class="op">&lt;-</span> <span class="va">project</span><span class="op">$</span><span class="fu">plotSNPs</span><span class="op">(</span></span>
<span>  <span class="st">'JAK1'</span>,</span>
<span>  group.by <span class="op">=</span> <span class="st">'snpGrade'</span>,</span>
<span>  split.by <span class="op">=</span> <span class="st">'patient_donor'</span>,</span>
<span>  min_alt_frac <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  data_out <span class="op">=</span> <span class="cn">TRUE</span>             <span class="co"># Return data frame instead of plot</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Save to CSV for external analysis</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">jak1_data</span>, <span class="st">"JAK1_SNPs.csv"</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Andrew Potter.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
